From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LineageOS Maintainer <maintainer@lineageos.org>
Date: Mon, 3 Aug 2026 22:45:00 +0700
Subject: [PATCH] telephony: Add writeUtil stub class for MediaTek IMS

Add com.android.internal.telephony.writeUtil stub class to prevent
NoClassDefFoundError crashes in com.mediatek.ims during call disconnect.

Signed-off-by: LineageOS Maintainer <maintainer@lineageos.org>
---
 .../com/android/internal/telephony/writeUtil.java  | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 src/java/com/android/internal/telephony/writeUtil.java

diff --git a/src/java/com/android/internal/telephony/writeUtil.java b/src/java/com/android/internal/telephony/writeUtil.java
new file mode 100644
index 000000000..111111111
--- /dev/null
+++ b/src/java/com/android/internal/telephony/writeUtil.java
@@ -0,0 +1,13 @@
+package com.android.internal.telephony;
+
+import android.content.Context;
+import android.util.Log;

+public class writeUtil {
+    private static final String LOG_TAG = "writeUtil";
+
+    public static void updateCauseInfo(int cause, int phoneId, Context context, boolean isVideo) {
+        Log.d(LOG_TAG, "updateCauseInfo: cause=" + cause + ", phoneId=" + phoneId + ", isVideo=" + isVideo);
+    }
+}
-- 
2.25.1
